src/main/java/de/zalando/tip/zalenium/proxy/TestingBotRemoteProxy.java

```diff
@@ -0,0 +1,114 @@
+    private static final String TESTINGBOT_CAPABILITIES_URL = "https://%s:%s@api.testingbot.com/v1/browsers";
+    private static final Logger logger = Logger.getLogger(TestingBotRemoteProxy.class.getName());
+    private static final String TESTINGBOT_CAPABILITIES_BK_FILE = "testingbot_capabilities.json";
+    private static final String TESTINGBOT_KEY = getEnv().getStringEnvVariable("TESTINGBOT_KEY", "");
+    private static final String TESTINGBOT_SECRET = getEnv().getStringEnvVariable("TESTINGBOT_SECRET", "");
+
+    public TestingBotRemoteProxy(RegistrationRequest request, Registry registry) {
+        super(updateTBCapabilities(request, String.format(TESTINGBOT_CAPABILITIES_URL, TESTINGBOT_KEY,
+                TESTINGBOT_SECRET)), registry);
+    }
+
+    @VisibleForTesting
+    private static RegistrationRequest updateTBCapabilities(RegistrationRequest registrationRequest, String url) {
+        JsonElement tbCapabilities = getCommonProxyUtilities().readJSONFromUrl(url);
+        try {
+            registrationRequest.getCapabilities().clear();
+            String userPasswordSuppress = String.format("%s:%s@", TESTINGBOT_KEY, TESTINGBOT_SECRET);
+            String logMessage = String.format("[TB] Capabilities fetched from %s", url.replace(userPasswordSuppress, ""));
+            if (tbCapabilities == null) {
+                logMessage = String.format("[TB] Capabilities were NOT fetched from %s, loading from backup file",
+                        url.replace(userPasswordSuppress, ""));
+                tbCapabilities = getCommonProxyUtilities().readJSONFromFile(TESTINGBOT_CAPABILITIES_BK_FILE);
+            }
+            logger.log(Level.INFO, logMessage);
+            return addCapabilitiesToRegistrationRequest(registrationRequest, tbCapabilities);
+        } catch (Exception e) {
+            logger.log(Level.SEVERE, e.toString(), e);
+            getGa().trackException(e);
+        }
+        return registrationRequest;
+    }
+
+    private static RegistrationRequest addCapabilitiesToRegistrationRequest(RegistrationRequest registrationRequest, JsonElement tbCapabilities) {
+        for (JsonElement cap : tbCapabilities.getAsJsonArray()) {
+            JsonObject capAsJsonObject = cap.getAsJsonObject();
+            DesiredCapabilities desiredCapabilities = new DesiredCapabilities();
+            desiredCapabilities.setCapability(RegistrationRequest.MAX_INSTANCES, 5);
+            desiredCapabilities.setBrowserName(capAsJsonObject.get("name").getAsString());
+            desiredCapabilities.setPlatform(getPlatform(capAsJsonObject.get("platform").getAsString()));
+            if (!registrationRequest.getCapabilities().contains(desiredCapabilities)) {
+                registrationRequest.addDesiredCapability(desiredCapabilities);
+            }
+        }
+        return registrationRequest;
+    }
+
+    private static String getBrowser(String browserName) {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Remove this unused private "getBrowser" method. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AUnusedPrivateMethod)